File: scripts/bin_script.py
from flask import Flask, render_template_string
import os

app = Flask(__name__)

# Define the file path to store the last scanned bin
file_path = 'last_scanned_bin.txt'

def save_last_scanned_bin(bin_id):
    """Save the last scanned bin ID to a file."""
    with open(file_path, 'w') as file:
        file.write(bin_id)
    print(f"Bin {bin_id} has been saved as the last scanned bin.")

def get_last_scanned_bin():
    """Retrieve the last scanned bin ID from the file."""
    if os.path.exists(file_path):
        with open(file_path, 'r') as file:
            bin_id = file.read().strip()
            return bin_id
    else:
        return None

@app.route('/')
def index():
    """Render the index page with the last scanned bin information."""
    last_bin = get_last_scanned_bin()
    if last_bin:
        message = f"The last scanned bin was: {last_bin}"
    else:
        message = "No bin has been scanned yet."
    return render_template_string('<h1>{{ message }}</h1>', message=message)

if __name__ == "__main__":
    # Run the Flask web application in debug mode
    app.run(debug=True)

File: scripts/bin_script.js
const express = require('express'); // Import the express module
const fs = require('fs'); // Import the file system module
const path = require('path'); // Import the path module

const app = express(); // Create an instance of express
const filePath = path.join(__dirname, 'last_scanned_bin.txt'); // Define the path to the file that stores the last scanned bin ID

// Function to save the last scanned bin ID to a file
function saveLastScannedBin(binId) {
    fs.writeFileSync(filePath, binId); // Write the bin ID to the file
    console.log(`Bin ${binId} has been saved as the last scanned bin.`); // Log a message to the console
}

// Function to get the last scanned bin ID from the file
function getLastScannedBin() {
    if (fs.existsSync(filePath)) { // Check if the file exists
        const binId = fs.readFileSync(filePath, 'utf-8').trim(); // Read the bin ID from the file and trim any whitespace
        return binId; // Return the bin ID
    } else {
        return null; // Return null if the file does not exist
    }
}

// Define a route for the root URL
app.get('/', (req, res) => {
    const lastBin = getLastScannedBin(); // Get the last scanned bin ID
    const message = lastBin ? `The last scanned bin was: ${lastBin}` : 'No bin has been scanned yet.'; // Create a message based on whether a bin ID was found
    res.send(`<h1>${message}</h1>`); // Send the message as an HTML response
});

// Start the server and listen on port 3000
app.listen(3000, () => {
    console.log('Server is running on http://localhost:3000'); // Log a message to the console
});